src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -71,75 +137,47 @@ public CarAge getCarAge() {
         return typeOfEngine;
     }
 
-    public void setTypeOfEngine(final TypeOfEngine typeOfEngine) {
-        cleanStepsAfterCurrent(3);
-        currentQuestion = Step.TYPE_OF_ENGINE;
+    public void setTypeOfEngineM1(final M1TypeOfEngine typeOfEngine) {
+        cleanStepsAfterCurrentM1Branch(3);
         this.typeOfEngine = typeOfEngine;
     }
 
-    public VolumeOfEngine getVolumeOfEngine() {
-        return volumeOfEngine;
+    public M1EngineVolume getM1EngineVolume() {
+        return m1EngineVolume;
     }
 
-    public void setVolumeOfEngine(final VolumeOfEngine volumeOfEngine) {
-        final int stepID = 4;
-        cleanStepsAfterCurrent(stepID);
-        currentQuestion = Step.VOLUME_OF_ENGINE;
-        this.volumeOfEngine = volumeOfEngine;
+    public void setM1Volume(final M1EngineVolume volumeOfEngine) {
+        cleanStepsAfterCurrentM1Branch(4);
+        this.m1EngineVolume = volumeOfEngine;
     }
 
-    public String getChatID() {
-        return chatID;
+    public void setM2Volume(final M2EngineVolume volumeOfEngine) {
+        cleanStepsAfterCurrentExceptM1Branch(2);
+        this.m2EngineVolume = volumeOfEngine;
     }
 
-    //method checks for last answered question and sends next step, which depends on current user's answers
-    public Step getNextStep() {
-        switch (currentQuestion) {
-            case TRANSPORT_TYPE -> {
-                if (transportType == TransportType.M1) {
-                    return Step.COUNTRY_ORIGIN;
-                }
-            }
-            case COUNTRY_ORIGIN -> {
-                return Step.OWNERS_TYPE;
-            }
-            case OWNERS_TYPE -> {
-                if (ownersType == OwnersType.PHYSICAL || countryOrigin == CountryOrigin.EAES) {
-                    return Step.CAR_AGE;
-                } else {
-                    return Step.TYPE_OF_ENGINE;
-                }
-            }
-            case TYPE_OF_ENGINE -> {
-                if (typeOfEngine == TypeOfEngine.GASOLINE) {
-                    return Step.VOLUME_OF_ENGINE;
-                } else {
-                    return Step.CAR_AGE;
-                }
-            }
-            case VOLUME_OF_ENGINE -> {
-                return Step.CAR_AGE;
-            }
-            case CAR_AGE, FAREWELL -> {
-                return Step.FAREWELL;
-            }
-            case null, default -> {
-                return Step.TRANSPORT_TYPE;
-            }
-        }
-        return Step.TRANSPORT_TYPE;
+    public void setTruckUnitType(final TruckUnitClass truckUnit) {
+        cleanStepsAfterCurrentExceptM1Branch(3);
+        this.truckUnitClass = truckUnit;
+    }
+
+    public TruckUnitClass getTruckUnitType() {
+        return this.truckUnitClass;
+    }
+
+    public String getChatID() {
+        return chatID;
     }
 
-    private void cleanStepsAfterCurrent(final int stepCleaner) {
+    private void cleanStepsAfterCurrentM1Branch(final int stepCleaner) {
```

Comment:
It's strange that we have two different methods `cleanStepsAfterCurrentM1Branch` and `cleanStepsAfterCurrentExceptM1Branch` instead of one `cleanStepsAfterCurrent `

Reply:
Now we have three of them :) 
I can try to rewrite it into one and to check what will happen.